### PR TITLE
Apply Gem Devise

### PR DESCRIPTION
**Pull request by Zama15**
The gem divise was added and implemented, making the model and views. The views were styled.

## Description

- What changes are being made?
Implemented the gem device
- Why are these changes necessary?
For accomplish the requirement of a session control in the application
- How have these changes been tested?
With the local server PUMA

## List of Changes

1. Add the gem in the Gem file
2. Run the commands to generate the migration, model and the views
3. Styled all the views generated such as
    - sign in
    - sign up
    - recovery password

## Additional Information
Mobile:
image
image
image

Desktop:
image
image
image

## Checklist

- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,5 +9,7 @@
   # Defines the root path route ("/")
   root 'home#index'
 
+  get 'index' => 'home#index'
```
Comment (juperameza) [inline config/routes.rb]: Why is this here for?

**Review by juperameza — CHANGES_REQUESTED** [review]

**Review by MauricioNC — APPROVED** [review]

File: app/assets/stylesheets/components/_form.scss [hunk 2]
```diff
@@ -0,0 +1,94 @@
+.form-wrap {
+    border-radius: 15px;
+    display: flex;
+    flex-direction: column;
+    height: 100%;
+
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_form.scss]: remove this extra line

File: app/assets/stylesheets/components/_form.scss [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    .form-container .links-container a:first-child {
+      margin-right: auto;
+    }
+
+    .form-container .links-container a {
+      font-size: 14px;
+      padding: 10px;
+      text-align: center;
+    }
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_form.scss]: indented correctly at two spaces

File: app/assets/stylesheets/components/_form.scss [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+@include desktop-lg {
+  .form-wrap {
+    .form-container {
+      width: 30%;
+    }
+
+    .form-container .field {
+      align-items: center;
+      align-items: flex-start;
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_form.scss]: here only apply the flex-start, remove the first one.

File: app/views/devise/passwords/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+<%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+  <%= f.hidden_field :reset_password_token %>
+
+  <div class="field">
+    <%= f.label :password, "New password" %><br />
+    <% if @minimum_password_length %>
+      <em>(<%= @minimum_password_length %> characters minimum)</em><br />
+    <% end %>
+    <%= f.password_field :password, autofocus: true, autocomplete: "new-password" %>
```
Comment (JuanequeX) [inline app/views/devise/passwords/edit.html.erb]: set a `required: true` to validate the presence of the password when the user do the submit.

File: app/views/devise/passwords/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<div class="form-wrap">
+  <div class="form-container">
+    <h2>¿Olvidaste tu contraseña?</h2>
+
+    <%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :post }) do |f| %>
+      <%= render "devise/shared/error_messages", resource: resource %>
+
+      <div class="field">
+        <%= f.label :email, "Correo electrónico:" %><br />
```
Comment (JuanequeX) [inline app/views/devise/passwords/new.html.erb]: remove any `br` label in your code

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/components/_form.scss (hunk 2 above)
Comment (Zama15) [reply app/assets/stylesheets/components/_form.scss]: done

File: config/routes.rb (hunk 1 above)
Comment (Zama15) [reply config/routes.rb]: modify the route root to link to the list_event and let the home index

File: app/assets/stylesheets/components/_form.scss (hunk 3 above)
Comment (Zama15) [reply app/assets/stylesheets/components/_form.scss]: done

File: app/assets/stylesheets/components/_form.scss (hunk 4 above)
Comment (Zama15) [reply app/assets/stylesheets/components/_form.scss]: done

File: app/views/devise/passwords/edit.html.erb (hunk 5 above)
Comment (Zama15) [reply app/views/devise/passwords/edit.html.erb]: done

File: app/views/devise/passwords/new.html.erb (hunk 6 above)
Comment (Zama15) [reply app/views/devise/passwords/new.html.erb]: done

**Review by juperameza — APPROVED** [review]
lgtm

**Review by JuanequeX — APPROVED** [review]
